File: Controls/SimplePanel.vb
Namespace Controls

    Public Class SimplePanel
        Inherits Panel

        Protected Overrides Function MeasureOverride(ByVal AvailableSize As Size) As Size
            Dim MaxWidth = 0.0
            Dim MaxHeight = 0.0

            For Each C As UIElement In Me.Children
                C.Measure(AvailableSize)
                Dim Sz As Size = C.DesiredSize

                If MaxHeight < Sz.Height Then
                    MaxHeight = Sz.Height
                End If
                If MaxWidth < Sz.Width Then
                    MaxWidth = Sz.Width
                End If
            Next

            Return New Size(MaxWidth, MaxHeight)
        End Function

        Protected Overrides Function ArrangeOverride(ByVal FinalSize As Size) As Size
            For Each C As UIElement In Me.Children
                C.Arrange(New Rect(FinalSize))
            Next

            Return FinalSize
        End Function

    End Class

End Namespace

File: WPF/Controls/SimplePanel.cs
using System.Windows;
using System.Windows.Controls;

namespace Ks.Common.Controls
{
    public class SimplePanel : Panel
    {
        protected override Size MeasureOverride(Size AvailableSize)
        {
            var MaxWidth = 0.0;
            var MaxHeight = 0.0;

            foreach (UIElement C in this.Children)
            {
                C.Measure(AvailableSize);
                var Sz = C.DesiredSize;

                if (MaxHeight < Sz.Height)
                    MaxHeight = Sz.Height;
                if (MaxWidth < Sz.Width)
                    MaxWidth = Sz.Width;
            }

            return new Size(MaxWidth, MaxHeight);
        }

        protected override Size ArrangeOverride(Size FinalSize)
        {
            foreach (UIElement C in this.Children)
                C.Arrange(new Rect(FinalSize));

            return FinalSize;
        }
    }
}
